proto/int32.go

```diff
@@ -33,10 +31,6 @@ func encodeInt32(b []byte, p unsafe.Pointer, flags flags) (int, error) {
 
 func decodeInt32(b []byte, p unsafe.Pointer, flags flags) (int, error) {
 	u, n, err := decodeVarint(b)
-	v := flags.int64(u)
-	if v < math.MinInt32 || v > math.MaxInt32 {
-		return n, fmt.Errorf("integer overflow decoding %v into int32", v)
-	}
-	*(*int32)(p) = int32(v)
+	*(*int32)(p) = int32(flags.int64(u))
```

Comment:
Removing the range check and truncating the integer value seems like it could create new problems.

At first sight, `-1` shouldn't have passed the `v < math.MinInt32 || v > math.MaxInt32` condition.

Would you be able to contribute a test that highlights the issue you ran into?